lte/gateway/c/core/oai/tasks/ngap/ngap_amf_handlers.c

```diff
@@ -2071,3 +2072,132 @@ status_code_e ngap_handle_paging_request(
 
   OAILOG_FUNC_RETURN(LOG_NGAP, rc);
 }
+status_code_e ngap_amf_handle_pduSession_modify_response(
+    ngap_state_t* state, const sctp_assoc_id_t assoc_id,
+    const sctp_stream_id_t stream, Ngap_NGAP_PDU_t* pdu) {
+  OAILOG_FUNC_IN(LOG_NGAP);
+  Ngap_PDUSessionResourceModifyResponse_t* container = NULL;
+  Ngap_PDUSessionResourceModifyResponseIEs_t* ie = NULL;
+  m5g_ue_description_t* ue_ref_p = NULL;
+  MessageDef* message_p = NULL;
+  gnb_ue_ngap_id_t gnb_ue_ngap_id = 0;
+  amf_ue_ngap_id_t amf_ue_ngap_id = 0;
+  int rc = RETURNok;
+  imsi64_t imsi64 = INVALID_IMSI64;
+  container = &pdu->choice.successfulOutcome.value.choice
+                   .PDUSessionResourceModifyResponse;
+  NGAP_FIND_PROTOCOLIE_BY_ID(Ngap_PDUSessionResourceModifyResponseIEs_t, ie,
+                             container, Ngap_ProtocolIE_ID_id_AMF_UE_NGAP_ID,
+                             true);
+  if (ie) {
+    asn_INTEGER2ulong(&ie->value.choice.AMF_UE_NGAP_ID,
+                      (uint64_t*)&amf_ue_ngap_id);
```

Comment:
Use reinterpret_cast wherever it is necessary

Reply:
Sure, but this file is C file